File: TennisObjects2/Model/PlayerEventSummaryList.cs
using System;
using System.Collections.Generic;
using System.Text;
using log4net;
using Csla;
using Bahns.Data;

namespace TennisObjects
{
	[Serializable()]
	public class PlayerEventSummaryList : ReadOnlyListBase<PlayerEventSummaryList, PlayerEventSummary>
	{

		private static ILog log = LogManager.GetLogger(typeof(PlayerEventSummaryList).ToString());

		#region Business Properties and Methods
		//public PlayerEventSummary item
		//{
		//    get { return (PlayerEventSummary)List.Item(index); }
		//}
		#endregion

		#region Contains
		//public bool Contains(PlayerEventSummary item)
		//{
		//    foreach (PlayerEventSummary child in List)
		//    {
		//        if (child.Equals(item))
		//        {
		//            return true;
		//        }
		//    }
		//    return false;
		//}
		#endregion

		#region Shared Methods
		public static PlayerEventSummaryList GetPlayerEventSummaryList(long PlayerID)
		{
			return (PlayerEventSummaryList)DataPortal.Fetch(new Criteria(PlayerID));
		}
		#endregion

		#region Constructors
		private PlayerEventSummaryList()
		{
			//AllowSort = true;
		}
		#endregion

		#region Criteria
		[Serializable()]
		private class Criteria
		{
			public long PlayerID;
			public Criteria(long PlayerID)
			{
				this.PlayerID = PlayerID;
			}
		}
		#endregion

		#region Data Access
		protected override void DataPortal_Fetch(object Criteria)
		{
			Criteria crit = (Criteria)Criteria;
			using (SafeDataReader dr = DbHelper.ExecuteReader("csla_get_player_summary_events", crit.PlayerID))
			{
				IsReadOnly = false;
				while (dr.Read())
					Add(PlayerEventSummary.GetPlayerEventSummary(dr));
				IsReadOnly = true;
			}
		}
		#endregion

	}
}

File: Objects/PlayerEventSummaryList.vb
<Serializable()> _
Public Class PlayerEventSummaryList
    Inherits Csla.ReadOnlyCollectionBase

    Private Shared log As ILog = LogManager.GetLogger(GetType(PlayerEventSummaryList).ToString())

#Region "Business Properties and Methods"
    Default Public ReadOnly Property item(ByVal index As Integer) As PlayerEventSummary
        Get
            Return CType(List.Item(index), PlayerEventSummary)
        End Get
    End Property
#End Region

#Region "Contains"
    Public Overloads Function Contains(ByVal item As PlayerEventSummary) As Boolean
        For Each child As PlayerEventSummary In List
            If child.Equals(item) Then
                Return True
            End If
        Next
        Return False
    End Function
#End Region

#Region "Shared Methods"
    Public Shared Function GetPlayerEventSummaryList(ByVal PlayerID As Long) As PlayerEventSummaryList
        Return CType(DataPortal.Fetch(New Criteria(PlayerID)), PlayerEventSummaryList)
    End Function
#End Region

#Region "Constructors"
    Private Sub New()
        AllowSort = True
    End Sub
#End Region

#Region "Criteria"
    <Serializable()> _
    Private Class Criteria
        Public PlayerID As Long
        Public Sub New(ByVal PlayerID As Long)
            Me.PlayerID = PlayerID
        End Sub
    End Class
#End Region

#Region "Data Access"
    Protected Overrides Sub DataPortal_Fetch(ByVal Criteria As Object)
        Locked = False
        Dim crit As Criteria = CType(Criteria, Criteria)
		Dim dr As SafeDataReader = New SafeDataReader(DataAccess.ExecReader("csla_get_player_summary_events", crit.PlayerID))
        Try
            While dr.Read()
                List.Add(PlayerEventSummary.GetPlayerEventSummary(dr))
            End While
        Finally
            dr.Close()
        End Try
    End Sub
#End Region

End Class
